DragonSlayer.cpp

```diff
@@ -1,20 +1,35 @@
 #include "DragonSlayer.h"
 #include "Dragon.h"
+#include "Utility.h"
 
-//DragonSlayer::DragonSlayer
+DragonSlayer::DragonSlayer(std::string name_, int hitPoints_, int armor_) : Character(hitPoints_, armor_, 4), name(name_)
+{
+    attackItem = std::unique_ptr<AttackItem> (new AttackItem());
```

Comment:
..If you're gonna reset it entirely here with a new unique_ptr.

Pick one or the other.